src/mocks/data.ts

```diff
@@ -363,17 +363,17 @@ export const mockUserToUpdate = {
   schoolOther: undefined,
 };
 
-export const buildMockStudent = <T extends number>(
-  id: T extends typeof mockUser.id ? `Student ID cannot be the same as the mockUser: ${typeof mockUser.id}` : T,
-) => {
-  if (id === mockUser.id) throw Error("A mock student cannot have the same ID as the mockUser");
+export const buildMockStudent = <T>(id: T) => {
+  if (typeof id === "number" && id === mockUser.id) {
+    throw Error("A mock student cannot have the same ID as the mockUser");
+  }
   return {
     givenName: "Test",
     familyName: `Student ${id}`,
     email: `test-student-${id}@test.com`,
-    dateOfBirth: 888888888888,
-    gender: id % 2 === 0 ? "MALE" : "FEMALE",
-    registrationDate: DAYS_AGO(50),
+    dateOfBirth: 888888888888 as unknown as Date,
+    gender: Math.random() < 0.5 ? "MALE" : "FEMALE",
```

Comment:
This isn't wrong, but it has some consequences, some positives and some negatives. The positive consequence is that introducing randomness in your tests may help to discover new bugs. The downside is that your tests are no longer deterministic, and they may start intermittently failing, and because you haven't set an easy way to make them reproducible (by allowing the developer to remove the randomness elements), then this could cause headaches in the future. Because of this, I would suggest reverting the changes and make the gender dependent on the ID.